components/authz-service/server/policy.go

```diff
@@ -167,7 +168,7 @@ func (s *policyServer) GetPolicy(
 	ctx context.Context,
 	req *api.GetPolicyReq) (*api.Policy, error) {
 
-	err := confirmRequiredID(req.Id, "policy")
+	err := validate.RequiredID(req.Id, "policy")
```

Comment:
Just a thought -- If we defined this as
```go
type hasID interface {
  GetId() string
}

func RequiredID(x hasID, y string) error {
    // ...
}
```

...we could here call `validate.RequiredID(req, "policy")`. Now, that doens't make much of a difference, but we could then also do the same for ID and name, like

```go
func RequiredIDandName(x interface { hasID, hasName }, y string) error
```

that would let us do `validate.RequiredIDandName(req, "thing")`  instead of `validate.RequiredIDandName(req.Id, req.Name, "thing")`.

What do you think?

We could also try to make us of field masks if we want to go protobuf-all-in. 🤔 

----

Taking things further, we could wire this into a services middleware, and write something like "with everything that has an ID, that ID is required" via
```go
if x, ok:= req.(HasID); ok {
  if err := validate.RequiredID(x, "???"); err != nil {  /* ... */ }
}
```
but we'd not know what to put into `???` easily, I suppose...